task_sdk/src/airflow/sdk/definitions/asset/__init__.py

```diff
@@ -451,11 +451,16 @@ def iter_dag_dependencies(self, *, source: str, target: str) -> Iterator[DagDepe
 
         :meta private:
         """
+        from airflow.models.asset import retrieve_asset_ids
+        from airflow.utils.session import create_session
+
+        with create_session() as session:
+            asset_id = str(retrieve_asset_ids(assets=[self], session=session)[0])
         yield DagDependency(
             source=source or "asset",
             target=target or "asset",
             dependency_type="asset",
-            dependency_id=self.name,
+            dependency_id=asset_id,
```

Comment:
I am looking at how we use this in `api-server`: https://github.com/apache/airflow/blob/main/airflow/api_fastapi/core_api/routes/ui/dependencies.py#L55

Right now, this would result in the graphj just rendering the id number. I think we should add a separate `label` as a new field on `DagDependency`. This would also allow us to use `dag_display_name` which would also be great.

Reply:
yes, that's also doable, but would like to confirm how we want to display each types of node?

Reply:
dag -> dag_display_name or dag_id
asset -> uri: ..., name: ...
asset alias -> name: ...
asset ref -> name or uri (if cannot resolved to anything) -> uri: ..., name:... (resolved to asset)

Reply:
All of those look good except i would drop the `uri: ` or `name: ` prefix and for assets and asset refs and only pass the name.

Reply:
Trigger or Sensor might need to include `task_display_name` or `task_id`

Reply:
`AssetUniqueKey.from_asset(Asset(name="abc", uri="test"))` <- this could be shorten into something else by implementing a __str__ or something similiar

Reply:
The REST API is built around `id` so we should keep it to that. Even if it means a DB lookup

Reply:
updated